chainstate/types/src/pos_randomness.rs

```diff
@@ -47,7 +47,7 @@ impl PoSRandomness {
 
     pub fn from_block(
         epoch_index: EpochIndex,
-        header: &BlockHeader,
+        block_timestamp: BlockTimestamp,
```

Comment:
The idea of passing the header is that we can choose "whatever we need from the header", since the function is `from_block`.

Reply:
We only require the timestamp from a header to calculate randomness. It's a more flexible and clear dependency because this way we can mine a block without the block itself by calculating `PoSData` and `BlockTimestamp` that will pass the check. 
Please take a look at `mine` function and how it's used - I think it's better this way.

Reply:
Alrighty... thought it'll make it harder in the future to include anything else.